server/core_authenticate.go

```diff
@@ -62,7 +62,7 @@ func AuthenticateApple(ctx context.Context, logger *zap.Logger, db *sql.DB, clie
 	// Existing account found.
 	if found {
 		// Check if it's disabled.
-		if dbDisableTime.Status == pgtype.Present && dbDisableTime.Time.Unix() != 0 {
+		if dbDisableTime.Valid && dbDisableTime.Time.Unix() != 0 {
```

Comment:
`disable_time` column is `NOT NULL` in schema, so all these checks can be replaced with simple `time.Time` argument  and emptines checked with `.IsZero()`

Reply:
`.IsZero()` and `.Unix() == 0` are not equivalent, I'd rather not touch this code as it's working as intended